dll/win32/shell32/shlexec.cpp

```diff
@@ -2594,14 +2576,9 @@ EXTERN_C HINSTANCE WINAPI WOWShellExecute(HWND hWnd, LPCSTR lpVerb, LPCSTR lpFil
 EXTERN_C void WINAPI
 OpenAs_RunDLLW(HWND hwnd, HINSTANCE hinst, LPCWSTR cmdline, int cmdshow)
 {
-    OPENASINFO info;
+    CCoInit ComStaInit;
```

Comment:
Is it needed to do it here, or would it be better to do that in SHOpenWithDialog instead?

Reply:
Ok then choose whatever is simpler. But if it needs to be moved in shopen... function, I think you may need to quickly check whether on windows (iirc it's an export?) that function expects com to be already initialized, or if it can cope without that (meaning it deals with the initialization itself).